openstack/clustering/v1/clusters/results.go

```diff
@@ -0,0 +1,127 @@
+package clusters
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"reflect"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// commonResult is the response of a base result.
+type commonResult struct {
+	gophercloud.Result
+}
+
+// CreateResult is the response of a Create operations.
+type CreateResult struct {
+	commonResult
+}
+
+type Cluster struct {
+	Config          map[string]interface{} `json:"config"`
+	CreatedAt       time.Time              `json:"-"`
+	Data            map[string]interface{} `json:"data"`
+	Dependents      map[string]interface{} `json:"dependents"`
+	DesiredCapacity int                    `json:"desired_capacity"`
+	Domain          string                 `json:"domain"`
+	ID              string                 `json:"id"`
+	InitAt          time.Time              `json:"-"`
+	MaxSize         int                    `json:"max_size"`
+	Metadata        map[string]interface{} `json:"metadata"`
+	MinSize         int                    `json:"min_size"`
+	Name            string                 `json:"name"`
+	Nodes           []string               `json:"nodes"`
+	Policies        []string               `json:"policies"`
+	ProfileID       string                 `json:"profile_id"`
+	ProfileName     string                 `json:"profile_name"`
+	Project         string                 `json:"project"`
+	Status          string                 `json:"status"`
+	StatusReason    string                 `json:"status_reason"`
+	Timeout         int                    `json:"timeout"`
+	UpdatedAt       time.Time              `json:"-"`
+	User            string                 `json:"user"`
+}
+
+func ExtractActionFromLocation(location string) (string, error) {
```

Comment:
I understand that this function is convenient to have, but since it doesn't directly involve the request being sent or the response being parsed, let's remove it. This kind of function could be added directly to the application which is importing Gophercloud.

Let me know if I'm wrong, though, and it'll have a larger purpose later on.

Reply:
Don't have a strong opinion on where this function should live. But ExtractingAction is going to be used frequently in order to query the status.

Reply:
Understood. I'm guessing the reason for this is to poll the Actions API to determine when a certain action has finished?

If so, and given the function in #873 was very similar, it might make sense to add a function (or functions) to https://github.com/gophercloud/utils which will parse the action ID (returned from any resource) and poll until a given status.

I wrote something like this a while ago and have been meaning to add it to the official utils repo:

* https://github.com/jtopjian/gophertools/blob/master/common.go
* https://github.com/jtopjian/gophertools/blob/master/openstack/compute/v2/tools/servers.go

The utils repo is meant to provide convenience functions which Gophercloud doesn't require to function but would otherwise be repeated in almost every situation.